### ISSUE TITLE
Training accuracy

### ISSUE THREAD

**Issue by lorenzoFabbri** [williamFalcon/pytorch-lightning#57]
I was wondering whether there is something like `validation_end` but for training (e.g., `training_end`). I want to compute the training accuracy at the end of each epoch. Thanks!

**Comment by williamFalcon** [comment]
you can use hooks
(on_epoch_end)

https://williamfalcon.github.io/pytorch-lightning/Trainer/hooks/

there's a typo in that doc... but you're looking for on_epoch_end

**Comment by lorenzoFabbri** [comment]
Thanks but I'm having some difficulties.
If I save the accuracy from each training step, how can I compute the average accuracy? `on_epoch_end ` does not take anything as input, so I cannot write something similar to `validation_end`. Thanks.

**Comment by expectopatronum** [comment]
@lorenzoFabbri Did you figure out how to do this? I am also trying to do this but don't know how...

**Comment by lorenzoFabbri** [comment]
Hi. I actually decided not to use this library for that project. So I cannot help you!

**Comment by expectopatronum** [comment]
@lorenzoFabbri Thanks a lot for the fast response!

**Comment by Yevgnen** [comment]
@lorenzoFabbri Do you have any suggestion for replacement?

**Comment by lorenzoFabbri** [comment]
@Yevgnen Hi. I haven't use this library in a long time! Perhaps thins have changed in the meanwhile so I cannot really help you. I plan on giving it another try in the next few weeks, though.

**Comment by Yevgnen** [comment]
@lorenzoFabbri Thanks for quick reply! May I ask which library you are using?😅

**Comment by williamFalcon** [comment]
we introduced a full callbacks system now! give it a shot.

**Comment by lorenzoFabbri** [comment]
@Yevgnen I actually used plain PyTorch. I didn't have to do fancy things, so I wrote my own functions for training and testing.
